dev/sparktestsupport/modules.py

```diff
@@ -459,6 +459,10 @@ def __hash__(self):
     dependencies=[hive, mllib],
     source_file_regexes=[
         "R/",
+        "sql/core/src/main/scala/org/apache/spark/sql/api/r/",
+        "core/src/main/scala/org/apache/spark/api/r/",
+        "mllib/src/main/scala/org/apache/spark/ml/r/",
+        "core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala",
```

Comment:
I was thinking we already run the tests on those changes by dependency detection logic here ...

Reply:
do you mean `dependencies=[hive, mllib]`?
possibly - I'm not seeing `core/src` perhaps that's implicit or is it the default set.. I thought it would be better to be explicit in case the modules change

Reply:
Yes, it is. If my understanding is correct, both dependencies include `sparkr` and if there is any file not matching with it, it sets `root` that runs everything.

How about fixing it later when this one causes actual problems? I gets the point of explicitly setting them but less sure because it adds lines but does not look fixing an (actual or potential) issue ..

Reply:
fair enough, reverted.